unittests/platformagnostictests.py

```diff
@@ -336,7 +336,41 @@ def test_editorconfig_match_path(self):
 
         for filename, pattern, expected in cases:
             self.assertTrue(match_path(filename, pattern) is expected, f'{filename} -> {pattern}')
-    
+
+    def test_format_invalid_config_key(self) -> None:
+        fd, fname = tempfile.mkstemp(suffix='.ini', text=True)
+        self.addCleanup(os.unlink, fname)
+
+        with os.fdopen(fd, 'w', encoding='utf-8') as handle:
+            print('not_an_option = 42', file=handle)
+
+        with self.assertRaises(MesonException):
+            Formatter(Path(fname), use_editor_config=False, fetch_subdirs=False)
+
+    def test_format_invalid_config_value(self) -> None:
+        fd, fname = tempfile.mkstemp(suffix='.ini', text=True)
+        self.addCleanup(os.unlink, fname)
+
+        with os.fdopen(fd, 'w', encoding='utf-8') as handle:
+            print('max_line_length = string', file=handle)
```

Comment:
Why are you using `print(..., file=f)` an not `f.write(...)`?

Reply:
Because it prints the line termination for me. This is more an habit than a necessity. I can change it if you want.

Reply:
I would like that yeah, it's a sort of "self documenting" thing in my mind: "use write() with file I/O, use print() with terminal I/O()".

I wish python files had a `writeln` method :(